executor/infoschema_reader.go

```diff
@@ -1819,23 +1819,22 @@ func (e *memtableRetriever) setDataForStatementsSummary(ctx sessionctx.Context,
 func (e *memtableRetriever) setDataForPlacementPolicy(ctx sessionctx.Context) error {
 	checker := privilege.GetPrivilegeManager(ctx)
 	is := infoschema.GetInfoSchema(ctx)
-	ruleBundles := is.RuleBundles()
 	var rows [][]types.Datum
-	for _, bundle := range ruleBundles {
+	if err := is.ForEachBundle(func(bundle *placement.Bundle) error {
```

Comment:
The operations inside this iteration may be slow, which blocks other operations. I prefer to clone it within a lock, and then iterate the cloned snapshot.

Reply:
Done